integration/apps/hpl_cpu/hpl_cpu.py

```diff
@@ -0,0 +1,155 @@
+        self.exe_path = os.path.join(benchmark_dir, 'hpl-2.3/bin/Linux_Intel64/xhpl')
+        self.NBs=384 # This is the recommended size for Intel Scalable Xeon family.
+        process_grid_ratios = {
+            1: {'P': 1, 'Q': 1},
+            2: {'P': 1, 'Q': 2},
+            4: {'P': 2, 'Q': 2},
+            8: {'P': 2, 'Q': 4},
+            16: {'P': 4, 'Q': 4}
+        }
+        if num_nodes not in process_grid_ratios:
+            raise RuntimeError(f"Number of nodes {num_nodes} is not defined for HPL.")
+        self.P = process_grid_ratios[num_nodes]['P']
+        self.Q = process_grid_ratios[num_nodes]['Q']
+
+        self.N = int(round(math.sqrt(dram_for_app / 8)))
+        self._cpu_per_rank = cores_per_node
+
+        print(f'DRAM reserved for APP: {dram_for_app} Cores for app: {cores_per_node} N={self.N}')
+
+    def get_bash_setup_commands(self):
+        input_file = textwrap.dedent(f'''
+        HPLinpack benchmark input file
+        Innovative Computing Laboratory, University of Tennessee
+        HPL.out      output file name (if any)
+        6            device out (6=stdout,7=stderr,file)
+        1            # of problems sizes (N)
+        {self.N}     Ns
+        1            # of NBs
+        {self.NBs}   NBs
+        0            PMAP process mapping (0=Row-,1=Column-major)
+        1            # of process grids (P x Q)
+        {self.P}     Ps
+        {self.Q}     Qs
+        16.0         threshold
+        1            # of panel fact
+        1            PFACTs (0=left, 1=Crout, 2=Right)1
+        1            # of recursive stopping criterium
+        4            NBMINs (>= 1)
+        1            # of panels in recursion
+        2            NDIVs
+        1            # of recursive panel fact.
+        1            RFACTs (0=left, 1=Crout, 2=Right)
+        1            # of broadcast
+        0            BCASTs (0=1rg,1=1rM,2=2rg,3=2rM,4=Lng,5=LnM)
+        1            # of lookahead depth
+        0            DEPTHs (>=0)
+        2            SWAP (0=bin-exch,1=long,2=mix)
+        64           swapping threshold
+        0            L1 in (0=transposed,1=no-transposed) form
+        0            U  in (0=transposed,1=no-transposed) form
+        1            Equilibration (0=no,1=yes)
+        8            memory alignment in double (> 0)
+        EOF
+        ''')
+        # KMP_HW_SUBSET: forces 1 thread per core to be used. Technically this
+        # should not be needed since --geopm-hyperthreads-disable is set below.
+        # OMP_NUM_THREADS: # number of cores per node. Technically this should
+        # not be needed since setting _cpu_per_rank will prompt the launcher to
+        # choose that many cores.
+        # MKL_NUM_THREADS: # matches with above, effective if linked against MKL.
+        env_file = textwrap.dedent(f'''
+        export KMP_HW_SUBSET=1t
+        export OMP_NUM_THREADS={self._cpu_per_rank}
+        export MKL_NUM_THREADS={self._cpu_per_rank}
+        EOF
```

Comment:
If these aren't needed, I'd remove them.  I'd explain in the README about the nuances on these and the interactions with the launcher.

Also, if MKL_NUM_THREADS is not something that's set by default anywhere (I don't believe it is), it might be something else we want to explore setting in the launcher for performance reasons in other apps.  What happens if MKL_NUM_THREADS is not set here?

Reply:
I had the first two just in case, I guess I can remove them. I will keep MKL_NUM_THREADS. MKL_NUM_THREADS is used by MKL libraries, that is how we communicate hpl how many threads per MPI it is supposed to create, I was told to use this by Nitin. Looks like it is intricate how MKL_NUM_THREADS is used by NKL libraries vs OMP_NUM_THREADS: https://stackoverflow.com/questions/13783190/openmp-and-mkl-threading

Reply:
I put a bunch of info in the README.md.